rfd/0229-agent-inventory-ui.md

```diff
@@ -0,0 +1,232 @@
+
+##### Example response:
+
+```json
+{
+  "instances": [
+    {
+      "id": "1",
+      "type": "instance",
+      "agent": {
+        "name": "server1",
+        "version": "16.1.3",
+        "services": ["ssh", "db", "desktop"],
+        "upgrader": {
+          "type": "unit",
+          "version": "v2",
+          "group": "group1"
+        }
+      }
+    },
+    {
+      "id": "2",
+      "type": "instance",
+      "agent": {
+        "name": "kube1",
+        "version": "16.5.4",
+        "services": ["kube"],
+        "upgrader": {
+          "type": "kube",
+          "version": "v2",
+          "group": "group1"
+        }
+      }
+    },
+    {
+      "id": "3",
+      "type": "bot",
+      "bot": {
+        "name": "bot1",
+        "version": "16.1.3",
+        "service": "kubernetes",
+        "health": "Unknown"
+      }
+    },
+    {
+      "id": "4",
+      "type": "instance",
+      "agent": {
+        "name": "agent1",
+        "version": "16.1.3",
+        "services": ["kube"],
+        "upgrader": {
+          "type": "kube",
+          "version": "v2",
+          "group": "group2"
+        }
+      }
+    }
+  ],
+  "startKey": ""
+}
+```
+
+#### RPC Service and Proto messages
+
+```protobuf
+service InventoryService {
+  // ListUnifiedInstances returns a page of teleport and bot instances. This API will refuse any requests when the cache is unhealthy or not yet
+  // fully initialized.
+  rpc ListUnifiedInstances(ListUnifiedInstancesRequest) returns (ListUnifiedInstancesResponse);
+}
+
+// ListUnifiedInstancesRequest is the request for listing instances.
+message ListUnifiedInstancesRequest {
+  // page_size is the size of the page to return.
+  int32 page_size = 1;
+  // page_token is the next_page_token value returned from a previous ListUnifiedInstances request, if any.
+  string page_token = 2;
+  // filters specify optional search criteria to limit which instances should be returned.
+  ListUnifiedInstancesFilter filter = 3;
+}
+
+// ListUnifiedInstancesResponse is the response from listing instances.
+message ListUnifiedInstancesResponse {
+  // items is the list of instances (teleport or bot instances) returned.
+  repeated UnifiedInstanceItem items = 1;
+  // next_page_token contains the next page token to use as the start key for the next page of instances.
+  string next_page_token = 2;
+}
+
+// ListUnifiedInstancesFilter provides a mechanism to refine ListUnifiedInstances results.
+message ListUnifiedInstancesFilter {
+  // search is a basic string search query which will filter results by name (hostname for teleport instances, bot name for bot instances).
+  string search = 1;
+  // advanced_search is an advanced search query using predicate language.
+  string advanced_search = 2;
+  // kinds is the kinds of instances to return. If ommitted, both teleport instances and bot instances will be returned.
+  repeated string kinds = 3;
```

Comment:
Suggestion: Rename this to instance_type. 

Reasoning: I think it would reduce confusion. ListUnifiedResources uses kinds to denote the teleport resources, which here are referred to as services.